### PR TITLE
Make the vscode clone link respect transport protocol (#20557)

### PR DESCRIPTION
**Pull request by ghost**
Backports #20557

### CODE REVIEW

File: templates/repo/home.tmpl [hunk 1]
```diff
@@ -126,7 +126,7 @@
 								<a class="item archive-link" href="{{$.RepoLink}}/archive/{{PathEscapeSegments $.RefName}}.tar.gz" rel="nofollow">{{svg "octicon-file-zip" 16 "mr-3"}}{{.i18n.Tr "repo.download_tar"}}</a>
 								<a class="item archive-link" href="{{$.RepoLink}}/archive/{{PathEscapeSegments $.RefName}}.bundle" rel="nofollow">{{svg "octicon-package" 16 "mr-3"}}{{.i18n.Tr "repo.download_bundle"}}</a>
 							{{end}}
-							<a class="item" href="vscode://vscode.git/clone?url={{$.RepoCloneLink.HTTPS}}">{{svg "gitea-vscode" 16 "mr-3"}}{{.i18n.Tr "repo.clone_in_vsc"}}</a>
+							<a class="item js-clone-url-vsc" href="vscode://vscode.git/clone?url={{$.RepoCloneLink.HTTPS}}">{{svg "gitea-vscode" 16 "mr-3"}}{{.i18n.Tr "repo.clone_in_vsc"}}</a>
```
Comment (silverwind) [inline templates/repo/home.tmpl]: Should this change to `.CloneButtonOriginLink` like original PR?

Comment (monim67) [reply templates/repo/home.tmpl]: @noerw can better answer it. I kept it unchanged.

Comment (noerw) [reply templates/repo/home.tmpl]: `.CloneButtonOriginLink` also exists in 1.17, I'd copy the line verbatim from the original PR.

Comment (noerw) [inline templates/repo/home.tmpl]:
```suggestion
							<a class="item js-clone-url-vsc" href="vscode://vscode.git/clone?url={{.CloneButtonOriginLink.HTTPS}}">{{svg "gitea-vscode" 16 "mr-3"}}{{.i18n.Tr "repo.clone_in_vsc"}}</a>
```

**Comment by monim67** [comment]
Is it ready to be merged now?

File: templates/repo/clone_script.tmpl [hunk 2]
```diff
@@ -19,5 +21,8 @@
 		for (const el of document.getElementsByClassName('js-clone-url')) {
 			el[el.nodeName === 'INPUT' ? 'value' : 'textContent'] = link;
 		}
+		for (const el of document.getElementsByClassName('js-clone-url-vsc')) {
+			el['href'] = 'vscode://vscode.git/clone?url=' + encodeURIComponent(link);
```
Comment (noerw) [inline templates/repo/clone_script.tmpl]: The linter wants to see this:
```suggestion
			el['href'] = `vscode://vscode.git/clone?url=${encodeURIComponent(link)}`;
```

Comment (monim67) [reply templates/repo/clone_script.tmpl]: Should we revert https://github.com/go-gitea/gitea/pull/20557/commits/fae64bf57e6795203fa2bd8b0e2621b04efcde2f then?

**Comment by monim67** [comment]
For me it's a bug, because I have HTTPS git disabled but clone in VSCode button always gives me HTTPS link. It never changes to SSH link. I had to override the template to replace HTTPS with SSH link to get it to work. I have to do this everytime a new version is released 😞

**Review by silverwind — APPROVED** [review]

**Comment by wxiaoguang** [comment]
@monim67-renforce : "This branch is out-of-date with the base branch": please enable Allow Edits from Maintainers, or update with 1.17 branch.

**Comment by monim67** [comment]
@wxiaoguang I cannot give write access as I created the PR from wrong account. None of the account has that allow edit checkbox now. I have updated the branch by merge.

**Comment by wxiaoguang** [comment]
I see. Just now there is a new commit in 1.17, could you update it again? Then we can get this merged in next time.

**Comment by monim67** [comment]
@wxiaoguang updated again.

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Another bug: Golang doesn't like the string syntax: ``` `...//...` ```, it will truncate the `//` as comment, then result in bad script code.

**Comment by wxiaoguang** [comment]
https://github.com/go-gitea/gitea/issues/21224#issuecomment-1254356110

**Pull request merged**